crates/integration_test/src/tests/cgroups/blkio.rs

```diff
@@ -0,0 +1,578 @@
+                .split_once(" ")
+                .context("invalid weight device format")?;
+            let (major_str, minor_str) = device_data
+                .split_once(":")
+                .context("invalid major-minor number format")?;
+            let leaf_weight: u16 = weight.parse().context("error in parsing leaf weight")?;
+            let major: i64 = major_str.parse().context("error in parsing major number")?;
+            let minor: i64 = minor_str.parse().context("error in parsing minor number")?;
+            let mut found = false;
+            for dev in &mut weight_devices {
+                if dev.major == major && dev.minor == minor {
+                    dev.leaf_weight = Some(leaf_weight);
+                    found = true;
+                }
+            }
+            if !found {
+                weight_devices.push(WeightDevice {
+                    major,
+                    minor,
+                    weight: None,
+                    leaf_weight: Some(leaf_weight),
+                });
+            }
+        }
+
+        device.weight_devices = weight_devices;
+    }
+
+    // throttle devices section -----
+
+    // we assume that if read_bps is supported, write_bps is also supported
+    if supports_throttle_bps() {
+        // throttle read bps
+        let throttle_read_bps_string =
+            fs::read_to_string(format!("{}/blkio.throttle.read_bps_device", path))
+                .context("error in reading block io write bps device")?;
+        let mut throttle_devices = Vec::new();
+        for line in throttle_read_bps_string.lines() {
+            let (device_data, rate) = line
+                .split_once(" ")
+                .context("invalid weight device format")?;
+            let (major_str, minor_str) = device_data
+                .split_once(":")
+                .context("invalid major-minor number format")?;
+            throttle_devices.push(ThrottleDevice {
+                major: major_str.parse().context("error in parsing major number")?,
+                minor: minor_str.parse().context("error in parsing minor number")?,
+                rate: rate
+                    .parse()
+                    .context("error in parsing throttle read bps rate")?,
+            });
+        }
+        device.throttle_read_bps_devices = throttle_devices;
+
+        // throttle write bps
+
+        let throttle_write_bps_string =
+            fs::read_to_string(format!("{}/blkio.throttle.write_bps_device", path))
+                .context("error in reading block io write bps device")?;
+        let mut throttle_devices = Vec::new();
+        for line in throttle_write_bps_string.lines() {
+            let (device_data, rate) = line
+                .split_once(" ")
+                .context("invalid weight device format")?;
+            let (major_str, minor_str) = device_data
+                .split_once(":")
+                .context("invalid major-minor number format")?;
```

Comment:
This snippet seems to be used a few times. Could be worth creating a method for this.